docs/source/daml/intro/8_Upgrading.rst

```diff
@@ -0,0 +1,170 @@
+#. In the same directory, load the chapter 8 Asset project using ``daml new 8Assets --template daml-intro-8-assets``.
+#. In the same directory, load the chapter 8 Upgrade project using ``daml new 8Upgrade --template daml-intro-8-upgrade``.
+
+``8_Assets`` is an improved version of the chapter 7 model, and ``8_Upgrade`` is a DAML model used to upgrade an existing, running application of the chapter 7 model to the new improved ``8_Assets`` model. The upgraded model has one major new feature: Multi-signatory owners. Wherever we had ``owner : Party`` in chapter 7, we now have ``owner : [Party]`` or something similar. This changes almost all templates. We can't just use composition and add new templates to the ledger, we need to convert old ``Asset`` contracts into new ones. We will use this feature to explore control flow, typeclasses, functions, and the Standard Libary in chapters 9 and 10, but for now, we will concentrate on the Upgrading part.
+
+DAR, DALF, DAML-LF, and the Engine
+----------------------------------
+
+In :doc:`7_Composing` you already learnt a little about projects, DAML-LF, DAR files, and dependencies. In this chapter we will actually need to have dependencies from the upgrade project to the chapter 7 and 8 asset projects so it's time to learn a little more about all this.
+
+Let's have a look inside the DAR file of chapter 7. DAR files, like Java JAR files are just ZIP archives, but the DAML SDK also has a utility to inspect DARs out of the box:
+
+#. Navigate into the ``7Composing`` directory.
+#. Build using ``daml build -o assets.dar``
+#. Run ``daml damlc inspect-dar assets.dar``
+
+You'll get a whole lot of output. Under the header "DAR archive contains the following files:" you'll see that the DAR contains
+
+#. ``*.dalf`` files for the project and all its dependencies
+#. The original DAML source code
+#. ``*.hi`` and ``*.hie`` files for each ``*.daml`` file
+#. Some meta-inf and config files
+
+The first file is something like ``7Composing-1.0.0-887056cbb313b94ab9a6caf34f7fe4fbfe19cb0c861e50d1594c665567ab7625.dalf`` which is the actual compiled package for the project. ``*.dalf`` files contain DAML-LF, which is DAML's intermediate language. The file contents are a binary encoded protobuf message from the `daml-lf schema <https://github.com/digital-asset/daml/tree/master/daml-lf/archive>`_.  DAML-LF is evaluated on the Ledger by the DAML Engine, which is a JVM component that is part of tools like the IDE's Script runner, the Sandbox, or proper production ledgers. If DAML-LF is to DAML what Java Bytecode is to Java, the DAML Engine is to DAML what the JVM is to Java.
+
+Hashes and Identifiers
+----------------------
+
+The  ``*.hi`` and ``*.hie`` files are interface files, which tell the DAML compiler how to map the bytecode back to DAML types.
```

Comment:
I don’t think that’s really an accurate description of interface files. They don’t map anything back. They describe something like the ABI of the package but at a DAML level not at a DAML-LF level. Moreover, I would really consider this an irrelevant implementation and would just omit it (for data-dependencies neither of those files are used). `.hie` files are only used for IDE stuff like cross-package goto definition.